### DISCUSSION TITLE
More intuitive way to initialize properties to the same value

### DISCUSSION THREAD

**Discussion by igristov** [dotnet/csharplang#7688] [Language Ideas]
Havin a class
```
class T
{
    public int A { get; set; }
    public int B { get; set; }
}
```
It would be great to have the possibility to initialize it like this:
```
var t = new T { A = B = foo() };
```
or like this:
```
var t = new T { A = foo(), B = A };
```
instead of existing way:
```
var t = new T();
t.A = t.B = foo();
```
or worse:
```
var t = new T { A = foo(), B = foo() };
```
If not in the language then maybe the compiler can support such syntax?

**Comment by TahirAhmadov** [comment]
https://github.com/dotnet/csharplang/discussions/5588

https://github.com/dotnet/csharplang/discussions/5588#discussioncomment-1888795
